syntaxes/alex.YAML-tmLanguage

```diff
@@ -0,0 +1,75 @@
+fileTypes:
+      - x
+name: Haskell Alex
+patterns:
+  - include: "#comments"
+  - include: "#blocks"
+  - include: "#syntax"
+  - include: "#strings"
+repository:
+  comments:
+    patterns:
+      - name: comment.line
+        match: \-\-.*$
+      - name: comment.block
+        begin: "{-"
+        end: "-}"
+  blocks:
+    name: meta.embedded.block.haskell
+    begin: "{"
+    beginCaptures:
+      "0": { name: punctuation.block.begin.alex }
+    end: "}"
+    endCaptures:
+      "0": { name: punctuation.block.end.alex }
+    patterns:
+      - include: source.haskell
+  syntax:
+    patterns:
+      - name: keyword.operator.pragma.alex
+        match: \%\w+
+      - name: variable.parameter.alex
+        match: \$\w+
+      - name: entity.name.class.alex
+        match: \@\w+
```

Comment:
Perhaps these should be `entity.name.macro.character-set.alex` and `entity.name.macro.regular-expression.alex`?